File: java/src/main/java/com/lacombe/promo3/taxi/Arrivals.java
package com.lacombe.promo3.taxi;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.Comparator;

public class Arrivals {

    private Collection<Arrival> arrivals = new ArrayList<>();

    public ArrivalHour getLastArrivalTime() {
       return arrivals.stream().map( a -> a.getHour()).max(Comparator.naturalOrder()).orElse(null);
    }

    public void add(Arrival arrival) {
        arrivals.add(arrival);
    }

    public Collection<Arrival> getArrivals() {
        return Collections.unmodifiableCollection(arrivals);
    }
}

File: dotnet/Socrates/TaxiOrganization/Arrivals.cs
using System.Collections.Generic;
using System.Linq;

namespace Socrates.TaxiOrganization
{
    public class Arrivals
    {
        private IList<Arrival> arrivals = new List<Arrival>();

        public ArrivalHour GetLastArrivalTime()
        {
            return arrivals.Max(arrival => arrival.GetHour());
        }

        public void Add(Arrival arrival)
        {
            arrivals.Add(arrival);
        }

        public IList<Arrival> GetArrivals()
        {
            return new List<Arrival>(arrivals);
        }
    }
}
